source/MaterialXRenderGlsl/TextureBaker.cpp

```diff
@@ -180,11 +197,149 @@ void TextureBaker::writeBakedDocument(NodePtr shader, const FilePath& filename)
     writeToXmlFile(bakedTextureDoc, filename);
 }
 
-FilePath TextureBaker::generateTextureFilename(OutputPtr output)
+// Helper function to generate mtlx filename
+FilePath TextureBaker::generateTextureFilename(OutputPtr output, const string udim)
 {
     string outputName = createValidName(output->getNamePath());
-    string udimSuffix = _udim.empty() ? EMPTY_STRING : "_" + _udim;
+    string udimSuffix = udim.empty() ? EMPTY_STRING : "_" + udim;
+
     return FilePath(outputName + "_baked" + udimSuffix + "." + _extension);
 }
 
+// Helper function to initialize file search pat
+FileSearchPath initFileSearchPath()
+{
+    FilePath installSearchPath = FilePath::getModulePath().getParentPath();
+    FilePath devSearchPath = FilePath(__FILE__).getParentPath().getParentPath().getParentPath();
+    FileSearchPath filename = FileSearchPath(installSearchPath);
+    filename.append(installSearchPath);
+
+    if (!devSearchPath.isEmpty() && devSearchPath.exists())
+    {
+        filename.append(devSearchPath);
+        devSearchPath = devSearchPath / "libraries";
+        if (devSearchPath.exists())
+        {
+            filename.append(devSearchPath);
+        }
+    }
+    return filename;
+}
+
+// Helper function to initialize shader generation context
+GenContext initGenContext()
+{
+    GenContext genContext = GlslShaderGenerator::create();
+    genContext.getOptions().hwSpecularEnvironmentMethod = SPECULAR_ENVIRONMENT_FIS;
+    genContext.getOptions().hwDirectionalAlbedoMethod = DIRECTIONAL_ALBEDO_TABLE;
+    genContext.getOptions().hwShadowMap = true;
+    genContext.getOptions().targetColorSpaceOverride = "lin_rec709";
+    genContext.getOptions().fileTextureVerticalFlip = true;
+    return genContext;
+}
+
+// Helper function to determine which materials to bake from renderable paths
+StringVec getRenderablePaths(DocumentPtr& doc)
+{
+    StringVec renderablePaths;
+    std::vector<TypedElementPtr> elems;
+    std::vector<TypedElementPtr> materials;
+    findRenderableElements(doc, elems);
+
+    if (elems.empty())
+    {
+        return StringVec();
+    }
+    for (TypedElementPtr elem : elems)
+    {
+        TypedElementPtr renderableElem = elem;
+        NodePtr node = elem->asA<Node>();
+        if (node && node->getType() == MATERIAL_TYPE_STRING)
+        {
+            std::vector<NodePtr> shaderNodes = getShaderNodes(node, SURFACE_SHADER_TYPE_STRING);
+            if (!shaderNodes.empty())
+            {
+                renderableElem = shaderNodes[0];
+            }
+            materials.push_back(node);
+        }
+        else
+        {
+            ShaderRefPtr shaderRef = elem->asA<ShaderRef>();
+            TypedElementPtr materialRef = (shaderRef ? shaderRef->getParent()->asA<TypedElement>() : nullptr);
+            materials.push_back(materialRef);
+        }
+        renderablePaths.push_back(renderableElem->getNamePath());
+    }
+    return renderablePaths;
+}
+
+// Helper function to generate texture filenames
```

Comment:
Since this is a purely module-local function, I would move it to an anonymous namespace at the top of the file.  Here's an example of this pattern in MaterialXCore:

https://github.com/materialx/MaterialX/blob/master/source/MaterialXCore/Document.cpp#L18